nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -181,23 +183,65 @@ public IcebergTable table(TableIdentifier tableIdentifier) {
   }
 
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    getRef().checkMutable();
+
+    if (namespace.isEmpty()) {
+      throw new IllegalArgumentException("Creating empty namespaces is not supported");
+    }
+
     try {
-      getRef().checkMutable();
-      withReference(
-              getApi()
-                  .createNamespace()
-                  .namespace(org.projectnessie.model.Namespace.of(namespace.levels()))
-                  .properties(metadata))
-          .create();
-      refresh();
-    } catch (NessieNamespaceAlreadyExistsException e) {
-      throw new AlreadyExistsException(e, "Namespace already exists: %s", namespace);
-    } catch (NessieNotFoundException e) {
+      org.projectnessie.model.Namespace content =
+          org.projectnessie.model.Namespace.of(metadata, namespace.levels());
+      ContentKey key = content.toContentKey();
+
+      CommitMultipleOperationsBuilder commitBuilder =
+          api.commitMultipleOperations()
+              .commitMeta(NessieUtil.buildCommitMetadata("create namespace " + key, catalogOptions))
+              .operation(Operation.Put.of(key, content));
+
+      Tasks.foreach(commitBuilder)
+          .retry(5)
+          // conflict can be thrown when parent namespace doesn't exist;
+          // bad request can be thrown when namespace already exists.
+          .stopRetryOn(
+              NessieNotFoundException.class,
+              NessieConflictException.class,
+              NessieBadRequestException.class)
+          .throwFailureWhenFinished()
+          .onFailure((o, exception) -> refresh())
+          .run(
+              b -> {
+                Branch branch = b.branch((Branch) getRef().getReference()).commit();
+                getRef().updateReference(branch);
```

Comment:
I'm not sure this is actually effective. We stop retries on `NessieConflictException`, which is basically the only reason for refreshing a reference :thinking: 

Also, modern Nessie Servers will be able to automatically resolve non-conflicting changes since the last branch refresh.

I suppose it we only need to refresh the reference when we refresh metadata.

Reply:
But shouldn't we strive to keep the client up-to-date with the latest commit hash on that ref? If I remove this statement, wouldn't the client keep a stale ref with the old hash?

Reply:
I added a few more test cases and I believe it's better to update the reference when the commit succeeds. This reduces the likelihood of future conflicts by keeping the HEAD of the reference up-to-date.